src/Benchmarks/Middleware/JsonMiddleware.cs

```diff
@@ -50,7 +49,7 @@ public static class JsonMiddlewareExtensions
         public static IApplicationBuilder UseJson(this IApplicationBuilder builder) => builder.UseMiddleware<JsonMiddleware>();
     }
 
-    [JsonSourceGenerationOptions]
+    [JsonSourceGenerationOptions(PropertyNamingPolicy = JsonKnownNamingPolicy.CamelCase)]
```

Comment:
Why is this needed if we're using the instance of the serializer options from DI that's already configured to use this casing by default?

Reply:
Because we add this context to the options, and when it serializes JsonMessage this context is used which has the default of `null` naming policy. And the naming policy needs to match the main options in https://source.dot.net/#System.Text.Json/System/Text/Json/Serialization/JsonSerializerContext.cs,74

I of course could be doing something wrong, these Json APIs are way too complex.

Reply:
It is not strictly necessary, since serialization will ultimately conform to whatever configuration is passed by the relevant `JsonSerializerOptions`. The benefit of declaring here though is that any naming policies will be computed at compile time and incorporated in the fast-path method.